core/ui/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/core/ui/remote_images/ProfilePictureImageProvider.kt

```diff
@@ -0,0 +1,32 @@
+package de.tum.informatics.www1.artemis.native_app.core.ui.remote_images
+
+import android.content.Context
+import coil.request.ImageRequest
+
+interface ProfilePictureImageProvider {
```

Comment:
Maybe it would make more sense to generalize the CourseImageProvider to also support profile pictures. This class looks like code duplication for me.